test/integration/persistent_volumes_test.go

```diff
@@ -596,14 +596,121 @@ func TestPersistentVolumeMultiPVsPVCs(t *testing.T) {
 	deleteAllEtcdKeys()
 }
 
+// TestPersistentVolumeControllerStartup tests startup of the controller.
+// The controller should not unbind any volumes when it starts.
+func TestPersistentVolumeControllerStartup(t *testing.T) {
+	_, s := framework.RunAMaster(t)
+	defer s.Close()
+
+	framework.DeleteAllEtcdKeys()
+	defer framework.DeleteAllEtcdKeys()
+
+	objCount := getObjectCount()
+
+	const shortSyncPeriod = 2 * time.Second
+	syncPeriod := getSyncPeriod(shortSyncPeriod)
+	testClient, binder, watchPV, watchPVC := createClients(t, s, shortSyncPeriod)
+	defer watchPV.Stop()
+	defer watchPVC.Stop()
+
+	// Create *bound* volumes and PVCs
+	pvs := make([]*api.PersistentVolume, objCount)
+	pvcs := make([]*api.PersistentVolumeClaim, objCount)
+	for i := 0; i < objCount; i++ {
+		pvName := "pv-startup-" + strconv.Itoa(i)
+		pvcName := "pvc-startup-" + strconv.Itoa(i)
+
+		pvc := createPVC(pvcName, "1G", []api.PersistentVolumeAccessMode{api.ReadWriteOnce})
+		pvc.Annotations = map[string]string{"annBindCompleted": ""}
+		pvc.Spec.VolumeName = pvName
+		pvc.Status.Phase = api.ClaimBound
+		newPVC, err := testClient.PersistentVolumeClaims(api.NamespaceDefault).Create(pvc)
+		if err != nil {
+			t.Fatalf("Cannot create claim %q: %v", pvc.Name, err)
+		}
+		pvcs[i] = newPVC
+
+		pv := createPV(pvName, "/tmp/foo"+strconv.Itoa(i), "1G",
+			[]api.PersistentVolumeAccessMode{api.ReadWriteOnce}, api.PersistentVolumeReclaimRetain)
+		claimRef, err := api.GetReference(newPVC)
+		if err != nil {
+			glog.V(3).Infof("unexpected error getting claim reference: %v", err)
+			return
+		}
+		pv.Spec.ClaimRef = claimRef
+		newPV, err := testClient.PersistentVolumes().Create(pv)
+		if err != nil {
+			t.Fatalf("Cannot create volume %q: %v", pv.Name, err)
+		}
+		pvs[i] = newPV
+	}
+
+	// Start the controller when all PVs and PVCs are already saved in etcd
+	binder.Run()
+	defer binder.Stop()
+
+	// wait for at least two sync periods for changes. No volume should be
+	// Released and no claim should be Lost during this time.
+	timer := time.NewTimer(2 * syncPeriod)
```

Comment:
Each PV and PVC evaluated every sync period. Assuming that the PVs/PVCs are bound together, this evaluation checks that a PV still has corresponding PVC and a vice versa. When one of this pair is missing, PV gets Released or PVC gets Lost.

So, let's wait for two sync periods to make sure at least one check happened (and finished) for sure.